test/CMakeLists.txt

```diff
@@ -233,6 +234,8 @@ if (TILEDB_ARROW_TESTS)
   )
 endif()
 
+add_definitions(-DTILEDB_TESTS)
```

Comment:
The next version of the code needs more work on the build. Here are some things you'll need to do to get the logger ready to integrate into a test runner.

1. `add_definitions` is out. We don't want to define a global symbol. 
2. We want to confine changes of behavior to `Logger` and its implementation. In order to run the test at issue, we need some kind of behavior within the `Logger`, but that should be the extent of the modifications. We'll need one compiler-provided definition to distinguish. This can be done with [`set_source_files_properties`](https://cmake.org/cmake/help/latest/command/set_source_files_properties.html) on the single file `logger.cc`.
   a. The property to set is [`COMPILE_DEFINITIONS`](https://cmake.org/cmake/help/latest/prop_sf/COMPILE_DEFINITIONS.html).
   b. Caution. The command is to set a definition, not to add a definition. You'll want to get the definition first, append to the result, and then set the definition.
   c. Call the symbol `TILEDB_USE_TEST_LOGGER`. This is a specific instruction, not just ambient for all tests.
3. Forward declare `LoggerTestPolicy` in `logger.cc`, but don't define it.
4. Mimic the behavior in `dynamic_memory.h` to use the defined symbol to select either `LoggerNullPolicy` (always defined) or a `LoggerTestPolicy` (conditionally defined). Use the potentially-specialized definition to select what kind of logger is defined.
5. Define `LoggerNullPolicy' in `logger.cc` and `ImplementationClass<LoggerNullPolicy>` in `logger.cc`
6. Define the classes `LoggerTestPolicy` and `ImplementationClass<LoggerTestPolicy>` in a source file in the test directory.

What this will all do is to allow the test code to contain the definitions needed for the code to link correctly. When the test policy is defined, it will be necessary to link in a test file that's not part of the library source to get the logger class to link. It would be useful to write a CMake target to ensure this, along the lines for the `compile_*` link test targets for object libraries.